akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/users/MoveUserToGuestLobbyCmdMsgHdlr.scala

```diff
@@ -0,0 +1,80 @@
+      msg.header.userId
+    ) || liveMeeting.props.meetingProp.isBreakout) {
+      val reason = "No permission to move user to guest lobby."
+      PermissionCheck.ejectUserForFailedPermission(meetingId, msg.header.userId, reason, outGW, liveMeeting)
+    } else {
+
+      // Eject user from the conference
+      for {
+        userMoved <- RegisteredUsers.findWithUserId(msg.body.userMovedToGuestLobbyId, liveMeeting.registeredUsers)
+        movedByUser <- RegisteredUsers.findWithUserId(msg.body.movedToGuestLobbyBy, liveMeeting.registeredUsers)
+      } yield {
+        RegisteredUsers.setWaitingForApproval(liveMeeting.registeredUsers, userMoved, GuestStatus.WAIT)
+
+        for {
+          user <- Users2x.ejectFromMeeting(liveMeeting.users2x, userMoved.id)
+        } yield {
+          UsersApp.sendUserLeftMeetingToAllClients(outGW, meetingId, userMoved.id)
+          UsersApp.sendEjectUserFromSfuSysMsg(outGW, meetingId, userMoved.id)
+          Sender.sendDisconnectClientSysMsg(meetingId, userMoved.id, movedByUser.id, EjectReasonCode.MOVED_TO_GUEST_LOBBY, outGW)
+          if (user.presenter) {
+            UsersApp.automaticallyAssignPresenter(outGW, liveMeeting)
+          }
+        }
+        for {
+          voiceUser <- VoiceUsers.findWithIntId(liveMeeting.voiceUsers, userMoved.id)
+        } yield {
+          UsersApp.sendEjectUserFromVoiceToFreeswitch(
+            outGW,
+            liveMeeting.props.meetingProp.intId,
+            liveMeeting.props.voiceProp.voiceConf, voiceUser.voiceUserId
+          )
+        }
+
+        //Notify Moderator about new guest and register the user anew
+        for {
+          movedUser <- RegisteredUsers.findWithUserId(userMoved.id, liveMeeting.registeredUsers)
+        } yield {
+          val movedUserToRegister = MovedUserToGuestLobby(movedUser.id, movedUser.externId, movedUser.name,
+            movedUser.role, movedUser.authToken, movedUser.avatarURL,
+            movedUser.guest, movedUser.authed, GuestStatus.WAIT,
+            movedUser.excludeFromDashboard, false)
+
+          // When the moved user lands in the guest lobby with the same sessionToken, he should poll for its status.
+          // The status value is stored in bbb-common-web.
+          // If the user session, doesn't changes its status,
+          // the guest user will be redirected directly depending on the previous response from moderator.
+          val eventStatusChanged = MsgBuilder.buildStatusOfMovedUserChangedEvtMsg(meetingId, movedUser.id, GuestStatus.WAIT)
+          outGW.send(eventStatusChanged)
+
+          // The moved user should be registered anew,
+          // since once he is ejected from the conference and moved to the guest lobby,
+          // he is removed also from the data concerning the registered users.
+          val eventRegisterUserAnew = MsgBuilder.buildRegisterMovedUserAnewEvtMsg(meetingId, movedUserToRegister);
+          outGW.send(eventRegisterUserAnew)
+        }
+      }
```

Comment:
Although this might be the correct series of events to be dispatched and guarantee the reset of an approved guest user, this `for-yeld` nesting could be refactored in multiple methods to reduce some of it's complexity.